oracles/engine_subscribers.go

```diff
@@ -68,6 +69,8 @@ func (r filterResult) hasMatched() bool {
 // specSubscriptions wraps the subscribers (in form of OnMatchedOracleData) to
 // the OracleSpec.
 type specSubscriptions struct {
+	mu *sync.RWMutex
```

Comment:
Mutexes don't need to be point, their 0 value when non pointer is a perfectly fine working mutex.

Reply:
Yes, but in that case the receiver `specSubscription` was a value, not a pointer itself, so we needed a memory location in it, not another value to exchange between threads.
This made me see that some of the methods have pointer receivers and some values - I made them all now to have pointer receivers so the mutex could be used as value.
Edited in a force push here: https://github.com/vegaprotocol/vega/pull/5580/commits/fe9606ae5a9b2b07306944c22f435bdd37724d5c